src/main.py

```diff
@@ -169,9 +172,46 @@ def start_django(port=5000):
     logging.info("Starting server...")
     setup_logging(debug=False)
     initialize()
+    automatic_provisiondevice()
     start.callback(port, background=False)
 
 
+def automatic_provisiondevice():
+    from kolibri.core.device.utils import device_provisioned
+    from kolibri.dist.django.core.management import call_command
+    from kolibri.utils.conf import KOLIBRI_HOME
+
+    AUTOMATIC_PROVISION_FILE = os.path.join(KOLIBRI_HOME, "automatic_provision.json")
+
+    if not os.path.exists(AUTOMATIC_PROVISION_FILE):
+        return
+    elif device_provisioned():
+        return
+
+    try:
+        with open(AUTOMATIC_PROVISION_FILE, "r") as f:
+            logging.info("Running provisiondevice from 'automatic_provision.json'")
+            options = json.load(f)
+    except ValueError as e:
+        logging.error(
+            "Attempted to load 'automatic_provision.json' but failed to parse JSON:\n{}".format(
+                e
+            )
+        )
+        sys.exit(1)
```

Comment:
Although it is overall an unlikely scenario, do we want to exit in this case? Since it is a GUI app and we don't provide a feedback prompt, if something did go wrong, to the user it would look like the app just closed automatically with no indication as to why. Although not ideal, would it be a problem to allow them to go through the setup wizard in case of failure?

Reply:
Oh, oops! Indeed, that was an error :) Fixed in the newest commit.